api/src/utils/csv-utils/csv-config-validation.ts

```diff
@@ -0,0 +1,179 @@
+      const cell = worksheetRow[header];
+      const params = { cell, header, row: worksheetRow, rowIndex: i + 1 };
+
+      callback(params, staticHeaderConfig);
+    }
+  }
+};
+
+/**
+ * Validate the CSV static and dynamic headers against the CSV config.
+ *
+ * @param {WorkSheet} worksheet - The worksheet
+ * @param {CSVConfig} config - The CSV configuration
+ * @returns {CSVError[]} - The CSV errors
+ */
+export const validateCSVHeaders = (worksheet: WorkSheet, config: CSVConfig): CSVError[] => {
+  const csvErrors: CSVError[] = [];
+
+  const configUtils = new CSVConfigUtils(worksheet, config);
+
+  if (!configUtils.headers.length) {
+    return [{ rowIndex: 0, error: 'CSV is empty', solution: 'Add headers and data to CSV' }];
+  }
+
+  for (const [staticHeader, headerConfig] of Object.entries(config.staticHeadersConfig)) {
+    const worksheetHasStaticHeader = configUtils.staticHeaders.some((header) =>
+      [staticHeader, ...headerConfig.aliases].includes(header)
+    );
+
+    if (!worksheetHasStaticHeader) {
+      csvErrors.push({
+        rowIndex: 0,
+        error: 'CSV missing required header',
+        header: staticHeader,
+        solution: `Add header '${staticHeader}' to CSV`
+      });
+    }
+  }
+
+  if (!config.ignoreDynamicHeaders && !config.dynamicHeadersConfig && configUtils.dynamicHeaders.length) {
+    for (const unknownHeader of configUtils.dynamicHeaders) {
+      csvErrors.push({
+        rowIndex: 0,
+        error: 'Unknown header in CSV',
+        header: unknownHeader,
+        solution: `Remove header '${unknownHeader}' from CSV`
+      });
+    }
+  }
+
+  return csvErrors;
+};
+
+/**
+ * Validate the CSV cells against the CSV config.
+ *
+ * @param {WorkSheet} worksheet - The worksheet
+ * @param {CSVConfig} config - The CSV configuration
+ * @returns {CSVError[]} - The CSV errors
+ */
+export const validateCSVCells = (worksheet: WorkSheet, config: CSVConfig): CSVError[] => {
+  const csvErrors: CSVError[] = [];
+
+  _forEachCSVCell(worksheet, config, (params, staticHeaderConfig) => {
+    if (staticHeaderConfig?.validateCell) {
+      csvErrors.push(...staticHeaderConfig.validateCell(params));
+    }
+
+    if (!staticHeaderConfig && config.dynamicHeadersConfig?.validateCell) {
+      csvErrors.push(...config.dynamicHeadersConfig.validateCell(params));
+    }
+  });
+
+  return csvErrors;
+};
+
+/**
+ * Set the cell values for the CSV worksheet.
+ *
+ * @param {WorkSheet} worksheet - The worksheet
+ * @param {CSVConfig} config - The CSV configuration
+ * @returns {CSVRow[]} - The CSV rows
+ */
+export const setCSVCellValues = (worksheet: WorkSheet, config: CSVConfig): CSVRow[] => {
+  const rows: CSVRow[] = [];
+
+  _forEachCSVCell(worksheet, config, (params, staticHeaderConfig) => {
+    if (staticHeaderConfig && staticHeaderConfig.setCellValue) {
+      delete params.row[params.header];
+      params.row[staticHeaderConfig.staticHeader.toUpperCase()] = staticHeaderConfig.setCellValue(params);
```

Comment:
Just curious: any reason for deleting the value first, rather than just overriding it? Unless the key also needs replacing, not just the value?

Reply:
Yea both the key and value need to be replaced.